### PR TITLE
Getting Upgrade-Assistant ready for .NET 6 GA

### LINKED ISSUE #879: Upgrade Assistant is ready for .NET 6 GA
This issue will track all the things we need to do to get the tool and experience ready for .NET 6 GA:

- Rev the NuGet package version number to 1.0.x
- Update the Current, LTS, and Preview target version numbers (at GA, all 3 will be 6.0)

### PR DESCRIPTION
**Pull request by sunandabalu**
Fixes #879

### CODE REVIEW

**Review by twsouthwick — APPROVED** [review]

File: tests/tool/Integration.Tests/Integration.Tests.csproj [hunk 1]
```diff
@@ -11,6 +11,9 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="NuGet.Frameworks" Version="6.0.0-rc.258" Condition=" '$(TargetFramework)' == 'net6.0' "/>
```
Comment (twsouthwick) [inline tests/tool/Integration.Tests/Integration.Tests.csproj:15]: Can you reference the VSTest issue?

**Review by twsouthwick — CHANGES_REQUESTED** [review]

File: tests/tool/Integration.Tests/Integration.Tests.csproj (hunk 1 above)
Comment (twsouthwick) [reply tests/tool/Integration.Tests/Integration.Tests.csproj:15]: and explain why this is needed?

File: eng/Versions.props [hunk 2]
```diff
@@ -2,8 +2,8 @@
 <Project>
   <PropertyGroup>
     <!-- This repo version -->
-    <MajorVersion>0</MajorVersion>
-    <MinorVersion>3</MinorVersion>
+    <MajorVersion>1</MajorVersion>
```
[the remark this answers is not in the document]
Comment (sunandabalu) [reply eng/Versions.props]: I'll prep a separate PR for version rev.

File: tests/tool/Integration.Tests/E2ETest.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@ public E2ETest(ITestOutputHelper output)
         [InlineData("WpfSample/csharp", "BeanTrader.sln", "BeanTraderClient.csproj")]
 
         // Skip on theories has issues in XUnit: https://github.com/xunit/visualstudio.xunit/issues/266
-#if !NET6_0
+#if !NET5_0
```
[the remark this answers is not in the document]
Comment (sunandabalu) [reply tests/tool/Integration.Tests/E2ETest.cs:40]: I tried, was still seeing the issue of empty aspnet assembly reference leading to 0 diagnostics for System.Web Analyzer leading to them not getting removed in the 5.0 tests. So leaving this as is.

[the remark this answers is not in the document]
Comment (twsouthwick) [reply tests/tool/Integration.Tests/E2ETest.cs:40]: Did you see the same exception message? Is it looking for a specific NuGet.Frameworks assembly?

[the remark this answers is not in the document]
Comment (sunandabalu) [reply tests/tool/Integration.Tests/E2ETest.cs:40]: no, for 5.0 tests as I mentioned above it failed to create diagnostics for SystemWebAnalyzer leaving the using System.Web imports behind in the actual output. There was no exception regarding the NuGet.Frameworks. Although adding a conditional package reference for 5.0 threw the exception in 6.0 tests.
`<PackageReference` Include="NuGet.Frameworks" Version="6.0.0-rc.258" Condition=" '$(TargetFramework)' == 'net6.0' "/>
    <PackageReference Include="NuGet.Frameworks" Version="5.11.0" Condition=" '$(TargetFramework)' == 'net5.0' "/>`
Issue to track re-enabling the tests for 5.0 - https://github.com/dotnet/upgrade-assistant/issues/911

**Review by twsouthwick — APPROVED** [review]

**Pull request merged**